crates/relayer/src/chain/axon/monitor.rs

```diff
@@ -112,22 +118,21 @@ impl AxonEventMonitor {
         &mut self,
         latest_block_count: u64,
     ) -> Result<Vec<IbcEventWithHeight>> {
-        let contract = Arc::new(Contract::new(
-            self.contract_address,
-            Arc::clone(&self.client),
-        ));
+        let contract = Contract::new(self.contract_address, Arc::clone(&self.client));
         let restore_block_number = self
             .start_block_number
             .checked_sub(latest_block_count)
             .ok_or(Error::others(format!(
                 "latest_block_count {latest_block_count} exceeds start_block_number {}",
                 self.start_block_number
             )))?;
-        let event_filter = |event: &OwnableIBCHandlerEvents| {
+        let event_filter = |event: &ContractEvents| {
             matches!(
                 event,
-                OwnableIBCHandlerEvents::SendPacketFilter(_)
-                    | OwnableIBCHandlerEvents::WriteAcknowledgementFilter(_)
+                ContractEvents::SendPacketFilter(_)
+                    | ContractEvents::WriteAcknowledgementFilter(_)
+                    | ContractEvents::RegisterCellEmitterFilterFilter(_)
```

Comment:
* How and when are these events emitted?
* What about previously registered filters?
* What's the significance of storing these filters on chain? (What's the difference with e.g. storing them offchain in a db?)
* Are all cells that we are interested in guaranteed to be emitted? Are they guaranteed to be emitted before the relavant ibc event?